Search results / Configure related records type depending on template.

fxprunayre opened the pull request:
The search API provides the possibility to load related records for each hits. This allows to easily display them in search results but it also slow down a bit the search.

Currently only the `list.html` template was displaying them based on the following configuration:

```
resultViewTpls: ...
grid: {
  related: ["parent", "children", "services", "datasets"]
},
```
![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/b49dd142-425b-4b11-9cc0-1e1f7ac8063b)

but user may want to customize this depending on the result template.



This move the configuration to the `resultViewTpls` (and preserve previous mechanism)

```
resultViewTpls: ...
              {
                tplUrl:
                  "../../catalog/components/" +
                  "search/resultsview/partials/viewtemplates/list.html",
                tooltip: "List",
                icon: "fa-bars",
                related: ["parent", "children", "services", "datasets"]
              },
```

so that user can more easily define which relations to load during search depending on the result template used.



Optimize search to avoid loading field not rendered in a template (eg. table mode don't display abstract and contacts)

![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/2ed663ba-c69e-4270-b6f9-c6b31a40535f)


```
              {
                tplUrl:
                  "../../catalog/components/" +
                  "search/resultsview/partials/viewtemplates/table.html",
                tooltip: "Table",
                icon: "fa-table",
                source: {
                  exclude: ["resourceAbstract*", "Org*", "contact*"]
                }
```

josegar74 reviewed (commented):
Changes look fine, tested and works.

But I can't find how to configure he new related configuration per search template results. If not present, I think should be included.

fxprunayre commented:
> But I can't find how to configure he new related configuration per search template results. If not present, I think should be included.

Added @josegar74 

![image](https://github.com/geonetwork/core-geonetwork/assets/1701393/0d59280a-6768-4b27-b9d0-2c008ca86fd3)

Also updated missing translations.

fxprunayre commented:
> @fxprunayre something not critical, but defining a for a template an array of related types to load and later emptying the section, doesn't save the change. You need to set value to `[]`. Maybe the initial values can be set to `[]` so it's more clear?

Added default empty array `[]` in config.
